crates/next_version/src/version_increment.rs

```diff
@@ -99,10 +100,24 @@ impl VersionIncrement {
                 .any(|commit| commit.commit_type == CommitType::Feature)
         };
 
+        let is_there_a_custom_match = |regex_option: &Option<Regex>| {
+            if let Some(regex) = regex_option {
+                commits.iter().any(|commit| {
+                    if let CommitType::Custom(ref custom_type) = commit.commit_type {
+                        return regex.is_match(custom_type);
+                    }
+                    false
+                })
+            } else {
+                false
+            }
+        };
```

Comment:
This can be extracted into a function and related unit tests could be added.

Reply:
done 4ae22475e0431002bfeeb849c940992d7aeadb86